lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -1297,6 +1355,95 @@ class ConvertAtenScatterReduceTwoOp
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Location loc = op.getLoc();
+
+    // Step 1. Fetch Input to sort.
+    Value inputTensor = adaptor.getSelf();
+    auto inputType = inputTensor.getType().cast<RankedTensorType>();
+    unsigned inputRank = inputType.getRank();
+
+    // Step 2. Fetch dimension to perform sort in.
+    int64_t dim;
+    if (!matchPattern(op.getDim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only constant dim value is supported");
+    dim = toPositiveDim(dim, inputRank);
+    if (!isValidDim(dim, inputRank)) {
+      return rewriter.notifyMatchFailure(op, "dim is statically invalid");
+    }
+
+    // Step 3. Fetch the order of sorting.
+    bool descending;
+    if (!matchPattern(op.getDescending(), m_TorchConstantBool(&descending)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only constant descending value is supported");
+
+    // Step 4. Form a RankedTensorType with same shape as that of the input's
+    //         but with elemental type i64.
+    RankedTensorType indicesType =
+        RankedTensorType::get(inputType.getShape(), rewriter.getI64Type());
+
+    // Step 5. Generate indices tensor.
+    SmallVector<Value> dynDims;
+    for (unsigned i = 0; i < inputType.getRank(); i++) {
+      if (inputType.isDynamicDim(i)) {
+        dynDims.push_back(rewriter.create<tensor::DimOp>(loc, inputTensor, i));
+      }
+    }
+    Value initEmptyTensor = rewriter.create<tensor::EmptyOp>(
+        loc, inputType.getShape(), rewriter.getI64Type(), dynDims);
+    Value zeroConst = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getZeroAttr(rewriter.getI64Type()));
+    Value outVector =
+        rewriter.create<linalg::FillOp>(loc, zeroConst, initEmptyTensor)
+            .getResult(0);
+
+    SmallVector<AffineMap> indexingMaps = {
+        AffineMap::getMultiDimIdentityMap(inputRank, op.getContext())};
+    SmallVector<utils::IteratorType> iteratorTypes(
+        inputRank, utils::IteratorType::parallel);
+    Value indicesTensor =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, outVector.getType(), ValueRange{}, outVector, indexingMaps,
+                iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value index = b.create<linalg::IndexOp>(loc, dim);
+                  index = castIndexToInt64(b, loc, index);
+                  b.create<linalg::YieldOp>(loc, index);
+                })
+            .getResult(0);
+
+    // Step 6. Create TMTensor::SortOp.
+    SmallVector<Value> operands;
+    operands.push_back(inputTensor);
+    operands.push_back(indicesTensor);
+    SmallVector<Type> elementTypes;
+    elementTypes.push_back(inputType.getElementType());
+    elementTypes.push_back(indicesType.getElementType());
+
+    TMTensor::SortOp sortOp = NULL;
```

Comment:
nit: It is safer to have `createTMTensorSortOp` create the entire op and return the result rather than populate the result into a reference. This also makes the behavior consistent with what `rewriter.create<...>(...)` does.

I would remove the `sortOp` argument from the helper function and change the result type to `FailureOr<Value>` to allow you to return the result from the sortOp.